### ISSUE TITLE
Add support for Jacoco reports

### ISSUE THREAD

**Issue by blindenvy** [gotwarlost/istanbul#782]
We have a build pipeline setup in jenkins that consists of java and javascript code. We use jacoco to produce code coverage reports for our java code base. We also have javascript projects where we're using Karma to run our jasmine tests and produce code coverage reports. Karma-coverage currently supports output format cobertura which has a jenkins plugin. We currently must display two separate code coverage reports (one for java (jacoco) the other for javascript (cobertura)). It would be nice if we could have code coverage reports for our java code and javascript code in the same format so that we could merge them all into one report. Seeing that jacoco is the more popular code coverage tool for java these days I see many others having this same need.

Jacoco has published a [DTD](https://github.com/jacoco/jacoco/blob/master/org.jacoco.report/src/org/jacoco/report/xml/report.dtd) of their format.

**Comment by adrianfalleiro** [comment]
@blindenvy Do you know how jacoco calculates complexity?

**Comment by blindenvy** [comment]
You can read all about it in their documentation: http://www.eclemma.org/jacoco/trunk/doc/counters.html

**Comment by sivasadhu** [comment]
+1

**Comment by elgohr** [comment]
+1

**Comment by cguckes** [comment]
+1

**Comment by slk9** [comment]
+1

**Comment by sschwenker** [comment]
+1

**Comment by ginamako** [comment]
+1

**Comment by fathergoose** [comment]
+1

**Comment by dejewi** [comment]
+1

**Comment by robcornelius** [comment]
+1

**Comment by mrvicadai** [comment]
+1

**Comment by mingz** [comment]
+1

**Comment by chintan-patel** [comment]
+1

**Comment by jfreitas2015** [comment]
+1

**Comment by justinhorton3** [comment]
+1

**Comment by webben-de** [comment]
+1

**Comment by ketanonline** [comment]
I agree with @blindenvy We have even more complex Jenkins eco system having multiple technologies. Having multiple plugins for coverage purpose doesn't help in maintenance/reporting areas. It will be great to see jacoco format report. :)

**Comment by bjuergens** [comment]
besides jenkins, another use-case for jacoco-reports would be, that they can be consumed by webstorm. Or at least the official documentation implies that it does

> [...] select the necessary .ic (IntelliJ IDEA runner) or .exec (JaCoCo runner) file [...]

 https://www.jetbrains.com/help/idea/switching-between-code-coverage-suites.html#add-remove-coverage-suite
